keepsorted/block.go

```diff
@@ -390,8 +390,8 @@ func (b block) lessFn() cmpFunc[lineGroup] {
 	longestFirst := comparing(func(s string) int { return len(s) }).reversed()
 	prefixes := slices.SortedFunc(slices.Values(b.metadata.opts.PrefixOrder), longestFirst)
 
-	prefixOrder := comparing(func(lg lineGroup) int {
-		p, ok := b.metadata.opts.hasPrefix(lg.joinedLines(), slices.Values(prefixes))
+	prefixOrder := comparing(func(s string) int {
```

Comment:
Is there a reason to do it this way as opposed to just caching `joinedLines()`? 

This approach isn't by any means bad or anything, but I fear it limits us in the future. Because if you want to add more options for changing the sorting in between here, you will now have a `string` as input instead of a `lineGroup`. Having a `lineGroup` is ideal because we can store complex data into there that we found when we parsed the line groups (as we already do for block mode). I also think that line 428 is a good bit less readable now than it was before.

Reply:
The main reason is that caching `joinedLines()` is a little bit harder, IMO. `lineGroup`s are mutable, so we'd have to be careful to invalidate the cache when it gets changed. This approach avoids that entire class of bugs

Reply:
While I do kinda think we can make it work (especially since I'm not sure if we typically even modify line groups after calling `joinedLines()` for the first time), that's fair. We may want to someday still do that, and then we'd wanna revert this change.